pkg/resources/kafka/pod.go

```diff
@@ -156,7 +174,7 @@ fi
 								},
 							},
 						},
-					},
+					}, r.KafkaCluster.Spec.Envs...),
```

Comment:
If the user specifies one of the env vars from above (e.g. KAFKA_HEAP_OPTS) in `r.KafkaCluster.Spec.Envs` than the env var list passed to Kafka will contain duplicate env vars.  We should either not allow users to override those env vars via `r.KafkaCluster.Spec.Envs` as for those there is already field in Spec and that should be used or handle duplicates.

I'd opt for the former catching duplicate env vars at validation.

Reply:
Thank you.

I had this in my mind actually and that's one of the reasons why I put those settings to monitoringConfig in the first commit https://github.com/banzaicloud/kafka-operator/pull/352/commits/b6accf4a64fc80f0c2530a0134e92d2447ac6ba0 :D

But to be honest, I am not sure how to catch this at validation. Can you please give me some guidance or point me in the right direction before I write something nasty? :D I have checked the kubebuilder validation spec but I don't see there anything useful that I could use for []corev1.EnvVar?

Sorry, I am not experienced much with this :/

Reply:
for now let's ensure that there are no duplicate env vars and `brokerConfig.GetKafkaHeapOpts()`  has precedence over `Spec.Envs` for `KAFKA_HEAP_OPTS` respectively `brokerConfig.GetKafkaPerfJmvOpts()` for  `KAFKA_JVM_PERFORMANCE_OPTS`.